lib/pool.js

```diff
@@ -44,6 +46,7 @@ class Pool extends EventEmitter {
     this[kNeedDrain] = false
     this[kPending] = 0
     this[kConnected] = 0
+    this[kTLSSessionCache] = new TLSSessionCache()
```

Comment:
Why not just a `Map` here directly?

Reply:
Thats a good question. It was quite practical to be able to tweak the behavior of the cache (map vs simpler or more complicated) and compare the different implementations.

This class adds very little and I don't feel strongly in keeping it or removing it. Happy to remove it if you want.

Reply:
Please remove. I'm very keen on simplicity.